scripts/slime.gd

```diff
@@ -0,0 +1,32 @@
+extends Node2D
+
+const SPEED = 40
+var direction = 1
+var turnAround = false
+@onready var ray_r: RayCast2D = $RayCastRight
+@onready var ray_dr: RayCast2D = $RayCastDownRight
+@onready var ray_l: RayCast2D = $RayCastLeft
+@onready var ray_dl: RayCast2D = $RayCastDownLeft
+@export var flip_h = false
+
+func _ready():
+	# Allows for changing direction upon level start
+	if flip_h:
+		$AnimatedSprite2D.flip_h = true
+		direction = - 1
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta: float) -> void:
+	if $AnimatedSprite2D.frame in range(1,12):
+		position.x += direction * SPEED * delta
+		# Not checking every frame cause issues where the slime keeps going when they would have stopped. 
```

Comment:
i like the comment
i'm not worried about performance so let's keep checking every frame